chromadb/api/fastapi.py

```diff
@@ -270,6 +270,15 @@ def get_collection(
             metadata=resp_json["metadata"],
         )
 
+    @trace_method("SegmentAPI.get_collection_by_id", OpenTelemetryGranularity.OPERATION)
+    @override
+    def get_collection_by_id(
+        self,
+        id: UUID,
+        embedding_function: Optional[EmbeddingFunction] = ef.DefaultEmbeddingFunction(),
+    ) -> Collection:
+        raise NotImplementedError()
```

Comment:
Doesn't this need to be implemented ?

Reply:
We need to change the server to support this method?

Reply:
I was thinking private, but, now that I added `id` to `get_collection`, not adding support for this on the server-side will feel awkward. Will add to FastAPI server though I wonder whether we should support it as-is:

`GET /api/v1/collections/{collection_name}`
and

`GET /api/v1/collections/{collection_id}`

But that too feels strange. Perhaps we can add `?type=name|id` with default `type=name` just so we keed backward compatibility, wdyt?

Reply:
Well if we add this to BaseAPI, we need to add it to fast api client, which means that we need it on the server yeah.

The way i worked around this was by adding a new path `/collections` which takes name or id as a body and in the future we can extend that to also support where filters etc. We can make /collections/{collection_name} a deprecated path in the future and move to this /collections path.

Reply:
Using POST to GET a collection seems counter-intuitive in a RESTful context, but we don't have many options here to make it work both ways without causing confusion. So yeah, POST with a body that takes id or name is the way to go. We can keep the existing endpoint and add a deprecation warning.

Reply:
Yeah, I think the restful-ness of our API is already pretty meh, and I'm ok corrupting it a bit